tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -534,4 +499,74 @@ void initTorchFunctions(PyObject* module) {
   }
 }
 
+PyObject* handle_torch_function(PythonArgs &r, PyObject* args, PyObject* kwargs, PyTypeObject &torch_api) {
+  PyObject* torch_api_function =
+    PyObject_FastGetAttrString((PyObject*)&torch_api, const_cast<char*>(r.get_func_name().data()));
+    PyObject* ret;
+    // there must be at least one overloaded argument at this point, since r.has_torch_function() is true
+    // so ret will never be returned unset
+    for (auto it = r.overloaded_args.begin(); it != r.overloaded_args.end(); ++it) {
```

Comment:
Kind of wondering why a range-for loop wasn't used here.

Reply:
Because I didn't know that was a thing :)